website/pages/docs/connect/terminating_gateway.mdx

```diff
@@ -0,0 +1,132 @@
+to connect to a service proxied by a terminating gateway will need to add the external service as an upstream destination.
+
+## Terminating Gateway Configuration
+
+Terminating gateways are defined in service configuration files and registered with Consul like other services, with two exceptions.
+The first is that the [kind](/api/agent/service#kind) must be "terminating-gateway". Second,
+the terminating gateway service definition may contain a `Proxy.Config` entry just like a
+Connect proxy service, to define opaque configuration parameters useful for the actual proxy software.
+For Envoy there are some supported [gateway options](/docs/connect/proxies/envoy#gateway-options) as well as
+[escape-hatch overrides](/docs/connect/proxies/envoy#escape-hatch-overrides).
+
+-> **Note:** If ACLs are enabled, the terminating gateways must be registered with a token granting `service:write`
+for the gateway's service name **and** all linked services. These privileges will authorize the gateway
+to terminate mTLS connections on behalf of the linked services.
+
+Linking services to a terminating gateway is done with a `terminating-gateway`
+[configuration entry](/docs/agent/config-entries/terminating-gateway). This config entry can be applied via the 
+[CLI](/docs/commands/config/write) or [API](/api/config#apply-configuration).
+
+Gateways with the same name in Consul's service catalog are configured with a single configuration entry.
+This means that additional gateway instances registered with the same name will determine their routing based on the existing configuration entry.
+Adding replicas of a gateway that routes to a particular set of services requires running the
+[envoy subcommand](/docs/commands/connect/envoy#terminating-gateways) on additional hosts and specifying
+the same gateway name with the `service` flag.
+
+~> [Configuration Entries](/docs/agent/config-entries) are global in scope. A configuration entry for a gateway name applies
+   across all federated Consul datacenters. If terminating gateways in different Consul datacenters need to route to different
+   sets of services within their datacenter then the terminating gateways **must** be registered with different names.
+
+The services that the terminating gateway will proxy for must be registered with Consul. They must also be registered
+in the same Consul datacenter as the terminating gateway. Otherwise the terminating gateway will not be able to
+discover the services' addresses. These services can be registered with a local Consul agent.
+If there is no agent present, the services can be registered [directly in the catalog](/api/catalog#register-entity)
+by sending the registration request to a client or server agent on a different host.
+
+Since Consul's services must be associated with a node, direct registrations to the catalog must define a node as well.
```

Comment:
I'm having a little trouble trying to understand the underlying issue. Is it that Consul registers all agents in the catalog by the node name and therefor all node names must be unique? Is there something else that TG introduces to make the issue more complicated?

Reply:
Yes node names are unique. The main issue is that agents periodically query the catalog for services registered to their node name. If someone registers an external service to that same node name:
1. The agent will find out
2. It will see it has no record of that service locally
3. It will send a request to delete that service from the catalog so that the catalog reflects its local state

Reply:
"All services registered in the Consul catalog must be associated with a node, include services registered directly with tools like Consul ESM. Ensure that services registered directly are associated with a node that has a unique name. Consul's anit-entropy will (then summarize those steps)"